File: 2021-blind/Q1.py
import string
def solution(new_id):
    id1 = new_id.lower()
    
    step2_filter = string.digits + string.ascii_lowercase + '-_.'
    id2 = ''
    for c in id1:
        if c in step2_filter: id2 += c
    
    id3 = ''
    for c in id2:
        if c != '.': id3 += c
        else:
            if id3 and id3[-1] == '.': continue
            id3 += c

    id4 = id3.strip('.')
    
    id5 = id4[:]
    if not id5: id5='a'
    
    id6 = id5[:15]
    if id6[-1]=='.': id6 = id6[:-1]
    
    id7 = id6[:]
    while len(id7) < 3: id7 = id7 + id7[-1]
    return id7

File: 2021-blind/Q1.cpp
#include <bits/stdc++.h>
using namespace std;

string solution(string new_id) {    
    string id1;
    for(auto c: new_id) id1 += tolower(c);
    
    string step2_filter = "0123456789abcdefghijklmnopqrstuvwxyz-_.";
    string id2;
    for(auto c : id1){
        if(find(step2_filter.begin(), step2_filter.end(), c) != step2_filter.end())
            id2 += c;
    }
    
    string id3;
    for(auto c : id2){
        if(c != '.') id3 += c;
        else{
            if(!id3.empty() && id3.back() == '.') continue;
            id3 += c;
        }
    }
    
    string id4;
    for(int i = 0; i < id3.size(); i++){
        if((i == 0 || i+1 == id3.size()) && id3[i] == '.') continue;
        id4 += id3[i];
    }
    
    string id5 = id4;
    if(id5.empty()) id5 = "a";
    
    string id6 = id5.substr(0, 15);
    if(id6.back() == '.') id6.pop_back();
    
    string id7 = id6;
    while(id7.size() < 3) id7 += id7.back();
    
    return id7;
}
